lms/djangoapps/courseware/access.py

```diff
@@ -62,9 +63,41 @@
 )
 from courseware.access_utils import adjust_start_date, check_start_date, debug, ACCESS_GRANTED, ACCESS_DENIED
 
+from lms.djangoapps.ccx.models import CustomCourseForEdX
+
 log = logging.getLogger(__name__)
 
 
+def has_ccx_coach_role(user, course_key):
+    """
+    Check if user is a coach on this ccx.
+
+    Arguments:
+        user (User): the user whose descriptor access we are checking.
+        course_key (CCXLocator): Key to CCX.
+
+    Returns:
+        bool: whether user is a coach on this ccx or not.
+    """
+    # To avoid circular imports
+    from lms.djangoapps.ccx.views import CCXLocatorValidationException
+    if isinstance(course_key, CCXLocator):
```

Comment:
Is it possible to use [duck typing](https://en.wikipedia.org/wiki/Duck_typing) here, rather than an `isinstance` check?